### DISCUSSION TITLE
Swiper mobile problem?

### DISCUSSION THREAD

**Discussion by ritchiemarkov** [bootscore/bootscore#305] [Q&A]
Hello Basti, its ofc me once again.

I have purchased dark theme (awesome!) and from mobile i have noticed that slider is a bit wired in visualizzation. its regards hero and cards. 

in hero you can check here: 

https://www.northwestalps.com/

on mobile its a mess. also read more button is strange. idk but you.an see your self its kinda differs a lot from desktop version.
another url https://www.northwestalps.com/2022/10/19/bootscore-swiper-gallery/

please let me know kindly if i did something wrong. but i dont think. is that depends on image size or (p) text in preview?

in cards there is a strange behavior on categories: https://www.northwestalps.com/2022/09/18/componenti-sciolti/

 
they are long. idk why and its pretty strange for me, but im sure into your support. Maybe some other faced this problem.

fyi im using cache enabler wich works awesome with your theme. please reccomend it. Thank you a lot.

**Comment by crftwrk** [comment]
Hi Richard,

I converted this issue into a discussion because this are no bugs. Let me explain.

### Heroes

Hero templates have featured image as a background and title, excerpt and button overlays the image. It depends how long your title and excerpt are. You can write a custom excerpt in the backend or limit length globally in your child's `functions.php` (default WordPress is 55 words):

```php
function mytheme_custom_excerpt_length( $length ) {
  return 20; // Limit excerpt words amount to 20
}
add_filter( 'excerpt_length', 'mytheme_custom_excerpt_length', 999 );
```

However, this are just templates and you can (and should) easily override them in your child https://bootscore.me/documentation/bs-swiper/#Overriding_templates_via_theme and build your own sliders. Check this hero-fade slider https://themes.bootscore.me/bc-fashion/ for what you can do with templates.

### Cards

Same as above. All cards have the same height and all cards have the height of the one with the longest content. So, card height depends on title and excerpt length as well.

#### Category badges

In gallery plugin's CSS https://github.com/crftwrk/bs-swiper-gallery/blob/main/wpt-swiper-gallery/wpt-swiper-gallery.css you will find this in line `32`:

```css
.swiper-slide a {
  display: block;
  width: 100%;
  height: 100%;
}
```

![screenshot](https://user-images.githubusercontent.com/51531217/199951763-616af810-886c-43bc-bdcf-019abda149bf.png)


This affects all swipers, no matter if it's a gallery or a card/hero. To address this only to the gallery it should be changed to:

```css
.wpt-swiper-gallery .swiper-slide a {
  display: block;
  width: 100%;
  height: 100%;
}
``` 

![screenshot-1](https://user-images.githubusercontent.com/51531217/199951882-d856d24e-6fd4-40b5-b9e7-3c7187e7c125.png)


If you want to create a PR, we can solve this quickly.

**Reply by ritchiemarkov** [reply]
Basti, first at all thank you for your detailed answer and your time. 
Well yes, i got something by my self about lenght of <p> into post description.
I have to play around right now with your suggestion.
Btw, this themes are so good! Is it your next project? Its looks awesome!

**Reply by crftwrk** [reply]
> Btw, this themes are so good! Is it your next project?

Of course, there should be some predesigned child-themes to purchase in the shop. But I doesn't have the time to finish them yet.

Do you want to create a PR for the gallery plugin to solve the badge bug? https://github.com/crftwrk/bs-swiper-gallery

**Reply by ritchiemarkov** [reply]
Yes of course im working on it now, and this evening gonna PR.
Sorry if im not mentionet it.
